packages/cli/src/commands/org/config/orgConfigBase.ts

```diff
@@ -23,7 +24,10 @@ export interface CustomFile extends ReadStream {
 export default abstract class SnapshotBase extends Command {
   public static description = 'Create and validate a snapshot';
 
-  public static flags = {
+  public static flags: {
+    target: IOptionFlag<string | undefined>;
+    deleteMissingResources?: IFlag<boolean>;
+  } = {
```

Comment:
Making `deleteMissingResource` flag optional, so children classes (such a Monitor) can benefit from the base class without extending all of its flags.

It's kinda hackish. However, another solution would be to simply not inherit from the `SnapshotBase` class and simply rewrite the required methods directly in the `Monitor` command.

Reply:
Yeah, that's the general problem with inheritance: you have to "nail" the abstraction in a very strict manner in order to not leak weird stuff into all children classes. 

And it's very hard to do it right on the first try and to never have to change it.

I would bite the bullet and refactor. Those kind of "add a hack in the parent class so that one of the child class is behave differently" is a sure sign that there's a flaw in the design, and they are prone to creating hard to debug problems.

If possible to use composition over inheritance, and to split the different "features" into separate class, and that you can then mix and match those classes in different command would make this much more robust.

For example, have different modules for the flag, another one for `dryRun`, `createSnapshotFromProject`, `validateSnapshot` etc.

And those different modules don't have to inherit from command: just simple functions/class.